### DISCUSSION TITLE
能否写一个禁用全站 P2P 上传下载的组件？

### DISCUSSION THREAD

**Discussion by Kaangnakamuyi** [the1812/Bilibili-Evolved#2901] [功能建议 / Ideas]
B 站的 P2P 从视频到直播都有，视频上的 P2P 节点一言难尽，冷门视频加载会卡顿，直播上的 P2P 会占用 CPU，也会导致卡顿，能否出一个禁用全站的 P2P 的组件？

**Comment by Geeyun-JY3** [comment]
直播的之前有人提过：#1585、#2364
#2364 打了个 `feat: rejected` 的 tag

**Reply by Kaangnakamuyi** [reply]
那视频的可以整吗（

**Reply by the1812** [reply]
https://github.com/the1812/Bilibili-Evolved/discussions/1585#discussioncomment-477452

你把他这个脚本的范围改一下, 视频页也 include 进去就行了

**Reply by cxw620** [reply]
> [#1585 (reply in thread)](https://github.com/the1812/Bilibili-Evolved/discussions/1585#discussioncomment-477452)
> 
> 你把他这个脚本的范围改一下, 视频页也 include 进去就行了

如果禁用, 有些视频会加载非常久, 会反复重试.  能不能发现采用了PCDN的视频加载就通过B站的接口重新解析视频获取普通CDN的视频呢.
此外观察接口返回数据发现返回的视频地址有备用的非PCDN的地址, 能不能直接拦截掉获取视频链接的API, 预处理一遍确保拿到了普通CDN的地址后再返回给播放器呢?

目前正在抓包分析. 目前发现卡顿只有网页端明显, 客户端只要屏蔽掉UPnP以及PCDN的域名就全走普通CDN了
